### ISSUE TITLE
Any plans for creating an in-depth React testing course?

### ISSUE THREAD

**Issue by konekoya** [coryhouse/ama#2]

**Comment by coryhouse** [comment]
I cover testing in Mocha with Chai in [React and Redux in ES6](https://app.pluralsight.com/library/courses/react-redux-react-router-es6) and [Building a JavaScript Development Environment](https://app.pluralsight.com/library/courses/javascript-development-environment).

I cover testing with Jest and Enzyme in [Creating Reusable React Components](https://app.pluralsight.com/library/courses/react-creating-reusable-components/table-of-contents).

That said, thanks for the suggestion! It's on my short list for sure. Please let me know if there some specific things you'd like to see covered.

**Comment by konekoya** [comment]
Hi Cory,
Sorry for the super late reply. We just celebrated The Lunar New Year last week.

I actually have watched all of the testing parts of your courses. They are really great, I Learned a lot from them. But still, I'm finding that when testing my own app, there's a lot of things that are hard to test or I just have no idea how to test and what should I test especially for testing UI.

I've read thru many articles and watched some great video courses like the one by Kent C Dodds
https://frontendmasters.com/courses/testing-javascript/ 

Still, no luck finding a comprehensive testing course that covers the things that I'm looking for.

Some more specific testings topics I can think of right now:
- Integrated testing and End-to-End testing for React (I'm not quite sure if they're different?)
- How to properly test UI logic, and what should be tested
- How to test async actions in Redux, I know you've covered this in one of your courses, but I think more examples on this topic would be great.

Basically, I think a course that dedicated to React testing and covers lots of different UI testing strategy would be really awesome.

I learned React and Redux with your Pluralsight courses. They're extremely helpful. Thanks, Cory, keep up your good work.
